src/libs/antares/writer/immediate_file_writer.cpp

```diff
@@ -83,62 +39,67 @@ ImmediateFileResultWriter::ImmediateFileResultWriter(const char* folderOutput):
 
 ImmediateFileResultWriter::~ImmediateFileResultWriter() = default;
 
-static bool prepareDirectoryHierarchy(const YString& root,
-                                      const std::string& entryPath,
-                                      Yuni::String& output)
+static bool prepareDirectoryHierarchy(const fs::path& root,
+                                      const fs::path& entryPath,
+                                      fs::path& output)
 {
-    output << root << Yuni::IO::Separator << entryPath.c_str();
-    return createDirectoryHierarchy(root, entryPath.c_str());
+    // Use relative path to remove root dir part
+    fs::path fullPath = root / entryPath.relative_path();
+    output = fullPath;
+
+    fs::path directory = fullPath.parent_path();
+    if (fs::exists(directory))
+    {
+        return true;
+    }
+
+    return fs::create_directories(directory);
 }
 
 // Write to file immediately, creating directories if needed
 void ImmediateFileResultWriter::addEntryFromBuffer(const std::string& entryPath,
                                                    Yuni::Clob& entryContent)
 {
-    Yuni::String output;
+    fs::path output;
     if (prepareDirectoryHierarchy(pOutputFolder, entryPath, output))
     {
-        IOFileSetContent(output, entryContent);
+        IOFileSetContent(output.string(), entryContent);
     }
 }
 
 // Write to file immediately, creating directories if needed
 void ImmediateFileResultWriter::addEntryFromBuffer(const std::string& entryPath,
                                                    std::string& entryContent)
 {
-    Yuni::String output;
+    fs::path output;
     if (prepareDirectoryHierarchy(pOutputFolder, entryPath, output))
     {
-        IOFileSetContent(output, entryContent);
+        IOFileSetContent(output.string(), entryContent);
     }
 }
 
-void ImmediateFileResultWriter::addEntryFromFile(const std::string& entryPath,
-                                                 const std::string& filePath)
+void ImmediateFileResultWriter::addEntryFromFile(const fs::path& entryPath,
+                                                 const fs::path& filePath)
 {
-    Yuni::String fullPath;
+    fs::path fullPath;
     if (!prepareDirectoryHierarchy(pOutputFolder, entryPath, fullPath))
     {
         return;
     }
 
-    switch (Yuni::IO::File::Copy(filePath.c_str(), fullPath))
+    std::error_code ec;
+    fs::copy(filePath, fullPath, ec);
```

Comment:
I think it would be simpler to use the message provided by the STL:
```cpp
try {
    fs::copy(filePath, fullPath);
} catch (const fs::filesystem_error& exc) {
    logs.error() << exc.what();
}
```

For example it provides such message:
```
filesystem error: cannot copy: No such file or directory [source] [target]
```
which is richer than just the error code message "No such file or directory".